### PR TITLE
Create a DTO for list user threads with tests

### PR DESCRIPTION
**Pull request by dominika-kokowicz**
Create a DTO for list user threads with tests

### CODE REVIEW

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        if let componentsURL = components.url?.absoluteString {
+            return componentsURL
+        } else {
+            return defaultPath
+        }
+    }
+    
+    let token: String
+    
+    var offset: Int = 0
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift:47]: Checkout `PaginatedRequest` on the main branch :)

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            
+            private enum CodingKeys: String, CodingKey {
+                case login
+                case avatarURL = "avatarUrl"
+            }
+        }
+        public let interlocutor: Interlocutor?
+    }
+    public let threads: [UserThread]
+    public let offset: Int
```
Comment (sloik) [inline App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift:28]: Checkout `PaginatedRequest` on `main` branch :)

**Review by sloik — APPROVED** [review]
Left some suggestions :)

**Comment by sloik** [comment]
Please rebase from main :D

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import Foundation
+import HTTPTypes
+
+// local
+import Onion
+
+/// https://developer.allegro.pl/documentation#operation/listThreadsGET
+public struct ListUserThreads: Codable, Equatable, ContentType {
+    
+    public struct UserThread: Codable, Equatable {
```
[the remark this answers is not in the document]
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift]: Fixed

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    var limit: Int = 20
```
[the remark this answers is not in the document]
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift]: Fixed

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        let queryItems: [(name: String, value: String?)] = [
+            ("limit", String(limit)),
+            ("offset", String(offset)),
+        ]
+        
+        components.queryItems?.append(contentsOf: queryItems.compactMap { name, value in
+            value.map { URLQueryItem(name: name, value: $0) }
+        })
+        
+        if let componentsURL = components.url?.absoluteString {
```
[the remark this answers is not in the document]
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift]: Fixed

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift (hunk 1 above)
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreadsRequest.swift]: Fixed :)

File: App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift (hunk 2 above)
Comment (dominika-kokowicz) [reply App/Lettuce/Sources/Lettuce/MessageCenter/ListMessages/GET/ListUserThreads/ListUserThreads.swift]: Fixed :)

**Pull request merged**